### PR TITLE
[Manta-PC] Add manta-xassets pallet(WIP)

### LINKED ISSUE #68: Test XCM message.

### PR DESCRIPTION
**Pull request by Dengjianping**
Closes #68 

Follow this wiki to test xcm: https://github.com/Manta-Network/Manta/wiki/Manta-Parachain-with-XCMP
Related commits:
polkadot: 91626111136243164792067392b110d84a129145
substrate: 87e63fa7c70be29383cb95ff52d71874f1db908a
cumulus: 540e74d33c09f27253f0f1c0671315fc035c371f

### CODE REVIEW

File: pallets/manta-xassets/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+	}
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(PhantomData<T>);
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T> {
+		// PhantomData<T>,
```
Comment (ghzlatarev) [inline pallets/manta-xassets/src/lib.rs]: Is this commented out code forgotten or still relevant ?

Comment (Dengjianping) [reply pallets/manta-xassets/src/lib.rs]: I forgot to remove it. Do it in next commit.

File: runtime/manta-pc/src/lib.rs [hunk 2]
```diff
@@ -465,7 +465,7 @@ pub type XcmOriginToTransactDispatchOrigin = (
 
 parameter_types! {
 	// One XCM operation is 1_000_000 weight - almost certainly a conservative estimate.
```
Comment (ghzlatarev) [inline runtime/manta-pc/src/lib.rs:492]: Update the comment, please.

File: pallets/manta-xassets/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+};
+
+pub use pallet::*;
+// Log filter
+const MANTA_XASSETS: &str = "manta-xassets";
+pub type BalanceOf<T> =
+	<<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+#[frame_support::pallet]
+pub mod pallet {
```
Comment (ghzlatarev) [inline pallets/manta-xassets/src/lib.rs:33]: So is this a new way of writing pallets, without macros ?

File: pallets/manta-xassets/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+				weight,
+				weight,
+			);
+			log::info! {target: MANTA_XASSETS, "xcm_outcome = {:?}", xcm_outcome};
+
+			Ok(().into())
+		}
+
+		/// transfer to relaychain
+		#[pallet::weight(10000)]
```
Comment (ghzlatarev) [inline pallets/manta-xassets/src/lib.rs:282]: We should log an issue to benchmark weights, or are they supposed to be constant 10k ?

File: runtime/manta-pc/src/lib.rs [hunk 5]
```diff
@@ -595,6 +660,33 @@ impl pallet_collator_selection::Config for Runtime {
 	type WeightInfo = pallet_collator_selection::weights::SubstrateWeight<Runtime>;
 }
 
+parameter_types! {
+	pub const MantaXassetsPalletId: PalletId = PalletId(*b"/ma_xast");
+
+	pub const AnyNetwork: NetworkId = NetworkId::Any;
+	pub FriendChains: Vec<(MultiLocation, u128)> = vec![
+		// Acala local and live, 0.01 ACA
```
Comment (ghzlatarev) [inline runtime/manta-pc/src/lib.rs:644]: What does this comment mean ?

File: pallets/manta-xassets/src/lib.rs (hunk 3 above)
Comment (Dengjianping) [reply pallets/manta-xassets/src/lib.rs:33]: Yes.

File: pallets/manta-xassets/src/lib.rs (hunk 4 above)
Comment (Dengjianping) [reply pallets/manta-xassets/src/lib.rs:282]: Now it's not our priority. We can do it later.

File: runtime/manta-pc/src/lib.rs (hunk 5 above)
Comment (Dengjianping) [reply runtime/manta-pc/src/lib.rs:644]: This is a tuple, the first element is which parachain we support, it likes a white list.
The second one is the min token to transfer while making cross chain transaction.

File: runtime/manta-pc/src/lib.rs (hunk 2 above)
Comment (Dengjianping) [reply runtime/manta-pc/src/lib.rs:492]: Will explain it next commit.

File: runtime/manta-pc/src/lib.rs (hunk 5 above)
Comment (ghzlatarev) [reply runtime/manta-pc/src/lib.rs:644]: Maybe just add this as a comment.

Comment (Dengjianping) [reply runtime/manta-pc/src/lib.rs:644]: Sure

**Review by ghzlatarev — APPROVED** [review]

**Pull request closed without merging**